lms/static/js/dateutil_factory.js

```diff
@@ -0,0 +1,102 @@
+
+/**
+ *
+ * A helper function to utilize DateUtils quickly in iterative display templates.
+ *
+ * @param: {string} data-datetime A pre-localized datetime string, assumed to be in UTC.
+ * @param: {string} data-timezone (optional) A user-set timezone preference.
+ * @param: {string} lang The user's preferred language.
+ * @param: {object} data-format (optional) a format constant as defined in DataUtil.dateFormatEnum.
+ *
+ * @param: {string} data-string (optional) a string for parsing through StringUtils after localizing
+ * datetime
+ *
+ * @return: {string} a user-time, localized, formatted datetime string
+ *
+ */
+
+(function(define) {
+    'use strict';
+
+    define([
+        'jquery',
+        'edx-ui-toolkit/js/utils/date-utils',
+        'edx-ui-toolkit/js/utils/string-utils'
+    ], function($, DateUtils, StringUtils) {
+    //     return function() {
+        var DateUtilIterator;
+        var localizedTime;
+        var stringHandler;
+        var displayDatetime;
+        var isValid;
+        var transform;
+
+        transform = function(iterationKey) {
+            var context;
+
+            $(iterationKey).each(function() {
+                if (isValid($(this).data('datetime'))) {
+                    context = {
+                        datetime: $(this).data('datetime'),
+                        timezone: $(this).data('timezone'),
+                        language: $(this).attr('lang'),
+                        format: $(this).data('format')
+                    };
+                    displayDatetime = stringHandler(
+                        localizedTime(context),
+                        $(this).data('string')
+                    );
+                    /*
+                    to show the prototype in action, check the console
+                    */
+                    /* eslint-disable no-alert, no-console */
+                    console.log(displayDatetime);
```

Comment:
As we just spoke, let's consider rolling this out as follows:
* Use this PR to merge in the low-level library and its own test
* No code would call this library in this PR.
* Your subsequent PRs would then update the various edx-platform code (including accordion) to call the new library.  Those PRs would remove old code and have tests to make sure the end-to-end is functioning as expected.

Given that, the console log statement can be replaced with the actual code (commented out below).